src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -7092,7 +7110,7 @@ DetachedPartsInfo MergeTreeData::getDetachedParts() const
         /// While it is possible to have detached parts on readonly/write-once disks
         /// (if they were produced on another machine, where it wasn't readonly)
         /// to avoid wasting resources for slow disks, avoid trying to enumerate them.
-        if (disk->isReadOnly() || disk->isWriteOnce())
+        if (disk->isReadOnly() || disk->isWriteOnce() || !lookOnDisk(disk))
```

Comment:
Perhaps move `disk->isReadOnly()` and `disk->isWriteOnce()` inside `lookOnDisk`?

Reply:
Actually it is an error, lookOnDisk (renamed to shouldSearchForPartsOnDisk) cannot be used here.

There are two different things:
1. detached parts (parts in a specific directory, that are not used)
2. parts on unexpected disks, that look like active parts

Fortunately there is no mixture and we are not looking for detached parts on unexpected disks.

This PR is a about (2) and I have to clean up the mess with e.g. parameter name.